extensions/libarchive/CompressContent.cpp

```diff
@@ -88,7 +88,7 @@ void CompressContent::processFlowFile(const std::shared_ptr<core::FlowFile>& flo
     std::string attr;
     flowFile->getAttribute(core::SpecialFlowAttribute::MIME_TYPE, attr);
     if (attr.empty()) {
-      logger_->log_error("No %s attribute existed for the flow, route to failure", core::SpecialFlowAttribute::MIME_TYPE);
+      logger_->log_error("No %s attribute existed for the flow, route to failure", core::SpecialFlowAttribute::MIME_TYPE.data());
```

Comment:
This is unsafe, unless you can guarantee that after the end of the pointed-to string span, there is a null byte. I'm not sure what the standard requires for string_views initialized from plain old string literals. ("Works on my compiler" is not safe enough.)

Reply:
you are right, good catch changed it in https://github.com/apache/nifi-minifi-cpp/pull/1612/commits/0b50c8b3c9a35f245d4a606d84f4fbd3634f54e5 
It would be good if our logger would accept string_view directly but thats for a new PR (maybe we could switch to fmt style aswell)

Reply:
based on the standard it should be safe as `std::string_view::data` returns `s` if constructed from `std::string_view::string_view(const char* s)`, but agree that playing it safe could not hurt

Reply:
If it's safe, then I'm fine with the original as well, I just wasn't sure

Reply:
I think your suggestion still makes sense, just wanted to check it if it is guaranteed or not

Reply:
I'll go with the explicit std::string ctor version (thats safe for sure) and we can revisit this when we add support for string_view in our logger (or when we switch it to fmt).